src/backend/d3d12/TextureD3D12.cpp

```diff
@@ -100,6 +100,20 @@ namespace d3d12 {
         device->GetResourceAllocator()->Release(resource);
     }
 
+    DXGI_FORMAT Texture::GetD3D12Format() const {
+        return D3D12TextureFormat(GetFormat());
+    }
+
+    uint32_t Texture::GetRowPitch() const {
+        uint32_t texelSize = 0;
+        switch (GetFormat()) {
+            case nxt::TextureFormat::R8G8B8A8Unorm:
+                texelSize = 4;
+        }
+        uint32_t rowSize = GetWidth() * texelSize;
+        return ((rowSize + (D3D12_TEXTURE_DATA_PITCH_ALIGNMENT - 1)) / D3D12_TEXTURE_DATA_PITCH_ALIGNMENT) * D3D12_TEXTURE_DATA_PITCH_ALIGNMENT;
```

Comment:
`((rowSize - 1) / D3D12_TEXTURE_DATA_PITCH_ALIGNMENT + 1) * D3D12_TEXTURE_DATA_PITCH_ALIGNMENT` is a little shorter. One of my CIS565 students pointed this out to me once. :P